tests/kernel/testImagesSetsUtilities.cpp

```diff
@@ -203,9 +203,11 @@ bool testSetFromImage()
   DigitalSet aSet5(d);
   DigitalSetInserter<DigitalSet> inserter5(aSet5);
   //predicate construction
-  typedef std::equal_to<Image::Value> EqualBinaryFunctor;
-  typedef std::binder2nd<EqualBinaryFunctor> ValuePredicate;
-  ValuePredicate equalTo1 (EqualBinaryFunctor(),1);
+  using ValuePredicate = std::function<bool(Image::Value)>;
+  ValuePredicate equalTo1 = [](Image::Value&& v)
+    {
+      return std::forward<Image::Value>(v) == 1;
```

Comment:
You can also simply pass `v` by const-ref and test `v == 1`. In fact, using `&&` here forbids lvalue parameters, like in the code
```
const auto fn = [] ( int&& v ) { return v == 1; };
int a = 3;
bool dummy = fn(a);
```
Here, since the `std::function` signature specifies a pass-by-value parameter, it doesn't fail.

Therefore, I also suggest to use a `std::function<bool(Image::Value const&)>`.

Reply:
Ahh, makes sense, still rusty with std::forward usage. I will change it in 12 hours